### PR TITLE
system: pam_limits: add ensure param

### PR DESCRIPTION
**Pull request by machacekondra**
Now you can specify 'ensure' param, which handle if line,
should exists or not in limits.conf file.

Issue: ansible#1083
Signed-off-by: Ondra Machacek machacek.ondra@gmail.com

### CODE REVIEW

File: system/pam_limits.py [hunk 1]
```diff
@@ -78,6 +78,12 @@
       - Modify the limits.conf path.
     required: false
     default: "/etc/security/limits.conf"
+  ensure:
```
Comment (bcoca) [inline system/pam_limits.py]: rename to state, that is the norm

**Comment by gregdek** [comment]
@usawa Please review this PR to make sure it adheres to the following guidelines:

http://docs.ansible.com/developing_modules.html#module-checklist

If it passes these guidelines, and if you believe it’s a good PR otherwise, please add a comment with "shipit" in the text, and we will flag it for inclusion.

If it fails these guidelines, please add a comment with "needs_revision" in the text, along with the needed changes, and we will flag it for revision.

Also, @uhawa, please submit a PR for this module with your Github ID in the author field so we can more easily identify you as the module owner here.

File: system/pam_limits.py [hunk 2]
```diff
@@ -78,6 +78,12 @@
       - Modify the limits.conf path.
     required: false
     default: "/etc/security/limits.conf"
+  state:
+    description:
+      - Ensure if record is in specified state
+    required: false
+    choices: [ "present", "absent" ]
+    default: "present"
```
Comment (abadger) [inline system/pam_limits.py:91]: Need to add `version_added: "2.0"`

**Comment by abadger** [comment]
Code looks good.  The one documentation thing I saw can be fixed now or after merging.

@evanccnyc did you test it worked?

**Comment by gregdek** [comment]
Thanks @machacekondra for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

**Comment by robynbergeron** [comment]
@machacekondra A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes.

**Comment by machacekondra** [comment]
ready_for_review

**Comment by robynbergeron** [comment]
Thanks @machacekondra. @usawa please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

**Comment by robynbergeron** [comment]
Thanks @machacekondra. @usawa please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

File: system/pam_limits.py (hunk 2 above)
Comment (abadger) [inline system/pam_limits.py:91]: Still need a version_added here.  It'll be 2.1 now.

**Comment by abadger** [comment]
I tested and found a problem.  state=absent toggles whether the line is present or not rather than removing the line from the file if found (otherwise doing nothing).

```
# Start without the line in /etc/security/limits.conf
[pts/1@roan /srv/ansible/ansible-friday/lib/ansible/modules/extras/packaging/language]$ ansible localhost -m pam_limits -a 'domain=badger limit_type=soft limit_item=nofile value=64000 state=absent' --sudo -K
SUDO password: 
localhost | SUCCESS => {
    "changed": true, 
    "msg": "badger\tsoft\tnofile\t64000\n"
}
# Now /etc/security/limits.conf has the line.

[pts/1@roan /srv/ansible/ansible-friday/lib/ansible/modules/extras/packaging/language]$ ansible localhost -m pam_limits -a 'domain=badger limit_type=soft limit_item=nofile value=64000 state=absent' --sudo -K
SUDO password: 
localhost | SUCCESS => {
    "changed": false, 
    "msg": "badger\tsoft\tnofile\t64000\n"
}
# Now the line has been removed from /etc/security/limits.conf again
```

changed=false is also being reported incorrectly.

**Comment by gregdek** [comment]
@usawa This change is still pending your review; do you have time to take a look and comment? Please comment with text 'shipit' or 'needs_revision' as appropriate.

**Comment by gregdek** [comment]
@usawa still waiting on your review.  Please comment with text 'shipit' or 'needs_revision' as appropriate. If we don't hear from you within 14 days, we will start to look for additional maintainers for this module.

**Comment by gregdek** [comment]
@machacekondra moving this to @ansible/core review. Thanks for your patience here.

**Pull request closed without merging**